packages/chevrotain/src/scan/lexer_public.ts

```diff
@@ -565,40 +565,44 @@ export class Lexer {
           // even though this pattern matched we must try a another longer alternative.
           // this can be used to prioritize keywords over identifiers
-          longerAltIdx = currConfig.longerAlt
-          if (longerAltIdx !== undefined) {
+          longerAlt = currConfig.longerAlt
+          if (longerAlt !== undefined) {
             // TODO: micro optimize, avoid extra prop access
             // by saving/linking longerAlt on the original config?
-            const longerAltConfig = patternIdxToConfig[longerAltIdx]
-            const longerAltPattern = longerAltConfig.pattern
-            altPayload = null
-
-            // single Char can never be a longer alt so no need to test it.
-            // manually in-lined because > 600 chars won't be in-lined in V8
-            if (longerAltConfig.isCustom === true) {
-              match = longerAltPattern.exec(
-                orgText,
-                offset,
-                matchedTokens,
-                groups
-              )
-              if (match !== null) {
-                matchAltImage = match[0]
-                if (match.payload !== undefined) {
-                  altPayload = match.payload
+            const longerAltLength = longerAlt.length
+            for (k = 0; k < longerAltLength; k++) {
+              const longerAltConfig = patternIdxToConfig[longerAlt[k]]
+              const longerAltPattern = longerAltConfig.pattern
+              altPayload = null
+
+              // single Char can never be a longer alt so no need to test it.
+              // manually in-lined because > 600 chars won't be in-lined in V8
+              if (longerAltConfig.isCustom === true) {
+                match = longerAltPattern.exec(
+                  orgText,
+                  offset,
+                  matchedTokens,
+                  groups
+                )
+                if (match !== null) {
+                  matchAltImage = match[0]
+                  if (match.payload !== undefined) {
+                    altPayload = match.payload
+                  }
+                } else {
+                  matchAltImage = null
                 }
               } else {
-                matchAltImage = null
+                this.updateLastIndex(longerAltPattern, offset)
+                matchAltImage = this.match(longerAltPattern, text, offset)
               }
-            } else {
-              this.updateLastIndex(longerAltPattern, offset)
-              matchAltImage = this.match(longerAltPattern, text, offset)
-            }
 
-            if (matchAltImage && matchAltImage.length > matchedImage.length) {
-              matchedImage = matchAltImage
-              payload = altPayload
-              currConfig = longerAltConfig
+              if (matchAltImage && matchAltImage.length > matchedImage.length) {
+                matchedImage = matchAltImage
+                payload = altPayload
+                currConfig = longerAltConfig
+                break
```

Comment:
This `break` statement is very important and is easy to miss.

please add a **comment** before the `break` statement that we exit the loop early
and the **first** longerAlt to match will take precedence.